src/wp-includes/formatting.php

```diff
@@ -2482,7 +2482,7 @@ function force_balance_tags( $text ) {
 	// Known single-entity/self-closing tags.
 	$single_tags = array( 'area', 'base', 'basefont', 'br', 'col', 'command', 'embed', 'frame', 'hr', 'img', 'input', 'isindex', 'link', 'meta', 'param', 'source' );
 	// Tags that can be immediately nested within themselves.
-	$nestable_tags = array( 'blockquote', 'div', 'object', 'q', 'span' );
+	$nestable_tags = array( 'blockquote', 'div', 'figure', 'object', 'q', 'span' );
```

Comment:
Only other comment I'd have is that this seems like it might be quite outdated. Should there be more added here to avoid future issues?

Things that might be missing:
- `<section>`
- `<article>` - the example here has immediate nesting https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article
- `<aside>`
- `<details>`

I didn't spend a great deal of time researching this, so open to suggestions otherwise. I may also have missed some tags that could be included here.

Reply:
> If it's definitely legacy, I wonder, why update it for the gallery block?

My understanding is although it is legacy, any older sites that have the `use_balanceTags` option set still filter content through `force_balance_tags` despite any database migrations. This was the case in our wider testing of the refactored gallery. For older sites that had this flag set the new gallery structure was broken on save, so it seems like the only way to fix this is to at least add `figure` as in this patch, or to remove the use of force_balance_tags for post content saving completely - not sure what the implications of that would be?

Reply:
To me, making this change for the gallery probably supports the argument that the tags here should be updated for other use cases. There might already be situations where users can directly nest these tags (custom or third-party blocks, Group block with the tag type option).